### PR TITLE
Add more components to 2.0.1 manifest and test manifests

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Add more components to 2.0.1 manifest and test manifests.
After this PR only two OpenSearch plugins (Security, Cross cluster replication) are missing because of https://github.com/opensearch-project/OpenSearch/issues/3554 with Core and security plugin. 

Two dashboards plugins need to be added later as well. 
https://github.com/opensearch-project/index-management-dashboards-plugin/pull/202
https://github.com/opensearch-project/security-dashboards-plugin/pull/1010

### Issues Resolved
Part of #2165 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.0.1/opensearch-2.0.1.yml [hunk 1]
```diff
@@ -52,3 +52,65 @@ components:
...
       - gradle:properties:version
       - gradle:dependencies:opensearch.version: notifications
+  - name: anomaly-detection
+    repository: https://github.com/opensearch-project/anomaly-detection.git
+    ref: '2.0'
+    checks:
+      - gradle:properties:version
+      - gradle:dependencies:opensearch.version
+  - name: opensearch-observability
+    repository: https://github.com/opensearch-project/observability
```
Comment (gaiksaya) [inline manifests/2.0.1/opensearch-2.0.1.yml:62]: Does this also needs to be a `.git` url like others?

Comment (zelinh) [reply manifests/2.0.1/opensearch-2.0.1.yml:62]: I think this shouldn't matter. I check previous few manifest and for `observability` plugin, the repository link is always with `.git` like others. I can add it if you think I have to.

Comment (gaiksaya) [reply manifests/2.0.1/opensearch-2.0.1.yml:62]: Lets keep it uniform and add it.
Thanks!

Comment (zelinh) [reply manifests/2.0.1/opensearch-2.0.1.yml]: Added. Please help review it now. Thanks.

**Review by gaiksaya — APPROVED** [review]
I think without security plugin we expect to see build and test failures

**Comment by zelinh** [comment]
> I think without security plugin we expect to see build and test failures

Security and CCR plugins have been added here! :D 
https://github.com/opensearch-project/security/pull/1881

**Comment by gaiksaya** [comment]
Ohh I read this and assumed they aren't 
```
After this PR only two OpenSearch plugins (Security, Cross cluster replication) are missing because of https://github.com/opensearch-project/OpenSearch/issues/3554 with Core and security plugin.
```

**Pull request merged**